### PR TITLE
[NFC][SYCL] Convert zero-dim-host-accessor and bfloat16_host LIT test to unittests

### PR DESCRIPTION
**Pull request by uditagarwal97**
Clone of https://github.com/intel/llvm/pull/8180

### CODE REVIEW

File: sycl/unittests/Extensions/BFloat16.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+#include <sycl/sycl.hpp>
+
+#include <gtest/gtest.h>
+
+#include <cmath>
+#include <limits>
+#include <string>
+
+namespace {
+// Helper to convert the expected bits to float value to compare with the result
```
Comment (maarquitos14) [inline sycl/unittests/Extensions/BFloat16.cpp:11]:
```suggestion
// Helper to convert the expected bits to float value to compare with the result.
```

**Review by maarquitos14 — APPROVED** [review]
LGTM. Just out of curiosity, why do we prefer these as unittests?

**Comment by uditagarwal97** [comment]
@intel/llvm-gatekeepers the PR is ready to be merged. E2E test failures is unrelated.

**Comment by martygrant** [comment]
hi @uditagarwal97 are you aware if these fails are being tracked already?

**Comment by AlexeySachkov** [comment]
> hi @uditagarwal97 are you aware if these fails are being tracked already?

@martygrant, that's a side effect of recent driver update. Those tests have been already addressed in #15823
Considering that this is an NFC patch, I will proceed with the merge

**Pull request merged**